tests/pass/tree_borrows/unique.rs

```diff
@@ -0,0 +1,40 @@
+//@revisions: default uniq
+//@compile-flags: -Zmiri-tree-borrows -Zmiri-tag-gc=0
+//@[uniq]compile-flags: -Zmiri-unique-is-unique
+
+#![feature(ptr_internals)]
+
+#[path = "../../utils/mod.rs"]
+mod utils;
+use utils::macros::*;
+
+use core::ptr::Unique;
+
+// Check general handling of Unique
+
+fn main() {
+    unsafe {
+        let base = &mut 5u8;
+        let alloc_id = alloc_id!(base);
+        name!(base);
+
+        let raw = &mut *base as *mut u8;
+        name!(raw);
+
+        // We create a `Unique` and expect it to have a fresh tag
+        // and uninitialized permissions.
+        let uniq = Unique::new_unchecked(raw);
+        let intermediate = if cfg!(uniq) { 2 } else { 0 };
```

Comment:
`intermediate`? I don't understand the name. This is about how many parents we need to walk up for to name the pointer, right?

In the macro docs this is called `nth_parent`, so maybe also use that name here, and explain why it needs an `if`.
(Also I just noticed in the macro sources it is called `nb`, whatever that means. Can you change that to `nth_parent` while we are at it?)